### PR TITLE
Rename EndianBuf to EndianSlice

### PR DESCRIPTION
**Pull request by fitzgen**
Makes it more clear that it isn't an owning pointer, and clears the way for adding an owned arc/rc/whatever version in #294

### CODE REVIEW

File: src/endian_slice.rs [hunk 1]
```diff
@@ -0,0 +1,362 @@
+//! Types for compile-time and run-time endianity.
```
Comment (philipc) [inline src/endian_slice.rs]: This needs updating.

File: src/endian_slice.rs [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+use parser::{Error, Result};
+use reader::Reader;
+
+/// A `&[u8]` slice with endianity metadata.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct EndianSlice<'input, Endian>
+where
+    Endian: Endianity,
+{
+    buf: &'input [u8],
```
Comment (philipc) [inline src/endian_slice.rs]: Worth renaming this too?

**Comment by fitzgen** [comment]
> I agree it's a better name, but not completely convinced its worth the churn for users of the library (they'll probably have their variables called buf still).

Did you see that there is a type alias `type EndianBuf = EndianSlice;` to ease upgrades? It is marked deprecated, with a note saying that the type has been renamed. Should hopefully mean that upgrading isn't actually very hard.

**Comment by fitzgen** [comment]
Which `rustfmt` have you been using? I just ran the latest nightly's `rustfmt-preview` component on `rustup` and it is introducing a ton of changes outside of the files I touched here.

**Comment by philipc** [comment]
The stable channel rustfmt-preview from rustc 1.25.

**Comment by philipc** [comment]
Don't know what's going on then. I'll run it when I rebase my PR.
```
$ rustc +stable --version
rustc 1.25.0 (84203cac6 2018-03-25)
$ cargo +stable fmt --version
0.3.8-nightly (346238f 2018-02-04)
```

**Pull request merged**